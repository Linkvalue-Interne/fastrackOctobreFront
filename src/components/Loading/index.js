import React from 'react';
import styled from '@emotion/styled';
import { keyframes } from '@emotion/core';

const loadingsping = keyframes`
100% {
  transform: rotate(360deg);
}`;

const LoadingSpinner = styled.div`
width: 2.5em;
height: 2.5em;
pointer-events: none;
position: fixed;
left: calc(50% + 1.25em);
top: calc(50%);
border: 0.4em solid transparent;
border-color: #eee;
border-top-color: #3E67EC;
border-radius: 50%;
animation: ${loadingsping} 1s linear infinite;
`;

const Loading = () => (
  <LoadingSpinner />
);

export default Loading;
