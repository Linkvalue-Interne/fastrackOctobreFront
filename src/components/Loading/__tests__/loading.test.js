import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';

import Loading from '../index';

expect.addSnapshotSerializer(serializer);

describe('Loading', () => {
  it('renders with correct styles', () => {
    const tree = renderer.create(<Loading />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
