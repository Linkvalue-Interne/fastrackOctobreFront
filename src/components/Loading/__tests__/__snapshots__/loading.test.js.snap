// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders with correct styles 1`] = `
@keyframes animation-0 {
  100% {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  width: 2.5em;
  height: 2.5em;
  pointer-events: none;
  position: fixed;
  left: calc(50% + 1.25em);
  top: calc(50%);
  border: 0.4em solid transparent;
  border-color: #eee;
  border-top-color: #3E67EC;
  border-radius: 50%;
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
}

<div
  className="emotion-0"
/>
`;
